tripal/tripal
# ChadoStorage::deleteValues()

# Tripal 4 Core Dev Task

### Issue #1350


### Tripal Version:  4.x

## Description


This PR completes the work for the `ChadoStorage::deleteValues()` function.  It supports removing data from Chado using the values of field properties and is currently designed to work when deleting an entity and the records in Chado need to be removed as well.   Simlar to the `findeValues()` it doesn't not work yet with any type of filtering (e.g. ranges, >=, etc.)


## Testing?



1. On a new site, create a contact entity
2. Create a project entity and associate the contact with it
3. If you want to compare unpublishing to deleting, feel free to unpublish the project. You should be able to publish it again and it will come back with as it was before with the contact linked to it.
4. Delete the project.
5. Clear the cache and reload the contact page. It should not have been deleted even though it was linked to the project.
6. Try to republish the project, it will not exist.
7. Run the following SQL statement to make sure the link in the `project_contact` table is gone: `select * from chado.project_contact`
